--- shapely/geos.py	2019-12-31 18:31:44.000000000 +0100
+++ shapely/geos.py	2020-03-23 08:48:10.430911180 +0100
@@ -221,7 +221,7 @@
                 # finishGEOS.
                 new_func.argtypes = [c_void_p]
             else:
-                new_func.argtypes = [c_void_p] + old_func.argtypes
+                new_func.argtypes = [c_void_p] + list(old_func.argtypes)
             if old_func.errcheck is not None:
                 new_func.errcheck = old_func.errcheck

